tools/CopyrightDateTool.scala

```diff
@@ -0,0 +1,171 @@
+import sys.process._
+
+object CopyrightDateTool {
+  val usage = """
+
+    This tool compares a copyright date in files with the date of last change in git.
+    If the message of last change contains "[Sync copyright dates]" it will be ignored and 
+    we will take next commit.
+
+    analyze - just shows if anything is ok
+    update - file fixing
+
+    From root folder:
+    Usage: scala tools/CheckDates.scala analyze
+           scala tools/CheckDates.scala update
+
+  """
+  trait DateComparison {
+    def fileLastUpdateYear: String
+    def gitLastUpdateYear: String
+    def fileCreationYear: String
+    def gitCreationYear: String
+
+    def isUpdateDateWrong(): Boolean = {
+      fileLastUpdateYear.toInt != gitLastUpdateYear.toInt
+    }
+
+    def isUpdateDateWrongStr(): Boolean = {
+      fileLastUpdateYear != gitLastUpdateYear
+    }
+  }
+
+  case class FileDates(pathToFile: String,
+                      fileLastUpdateYear: String,
+                      gitLastUpdateYear: String,
+                      fileCreationYear: String,
+                      gitCreationYear: String) extends DateComparison
+
+  case class GitInfo(message: String, date: String)
+
+  def readFilesList(): Array[String] = {
+    val git_files = Seq("/bin/sh", "-c", 
+      "git grep --name-only --fixed-strings" + 
+      " '|    (c) 20' -- '*.scala'|grep -v CopyrightDateTool").!!
```

Comment:
Today I learned
```
amarki@amarki-462836:~/projects/scala$ git grep -E -e '(\(c\)|Copyright) 20[0-9]{2}-(20[0-9]{2})' --and --not -e EPFL -- '*.scala' | more
src/library/scala/collection/mutable/History.scala:**    / __/ __// _ | / /  / _ |    (c) 2003-2013, LAMP/tPFL             **
```
I see you're checking only the banner here; it took me a while to understand the leading VBAR.

There are other badly formed or deviant notices, too, from third parties like the classic Contributor.